server/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala

```diff
@@ -86,8 +87,13 @@ private[omega_edit] class SessionImpl(p: Pointer, i: FFI) extends Session {
   def endTransaction: Int =
     i.omega_session_end_transaction(p)
 
-  def checkpointDirectory: Path =
-    Paths.get(i.omega_session_get_checkpoint_directory(p))
+  def checkpointDirectory: Path = {
+    val chkptDir: String = i.omega_session_get_checkpoint_directory(p).substring(0, this.checkpointDirectoryStrLen)
```

Comment:
It's probably worth a comment to describe _why_ this is unfortunately necessary.